web/ui/react-app/src/Panel.tsx

```diff
@@ -203,14 +193,21 @@ class Panel extends Component<PanelProps & PathPrefixProps, PanelState> {
 
   handleChangeEndTime = (endTime: number | null) => {
     this.setOptions({ endTime: endTime });
+    this.executeQuery();
   };
 
   handleChangeResolution = (resolution: number | null) => {
     this.setOptions({ resolution: resolution });
+    this.executeQuery();
+  };
+
+  handleChangeType = (type: PanelType) => {
+    this.setOptions({ type: type });
```

Comment:
I guess this is where another `this.executeQuery();` is needed as well.

Reply:
Good call. I will work on this

Reply:
@juliusv Switching between Graph/Table views now re-executes the query. I had to change a few things around. I think it's better now anyway.